### PR TITLE
Fix: Linkgraph legend assumes strings are small.

### PR DESCRIPTION
**Pull request by PeterN**
## Motivation / Problem

The linkgraph legend is designed for all legend strings to be in the small font, but doesn't enforce this.

image

## Description

Ensure all linkgraph legend strings are drawn with FS_SMALL forced, so that it does not rely on NewGRF authors.

image

## Checklist for review

Some things are not automated, and forgotten often. This list is a reminder for the reviewers.
* The bug fix is important enough to be backported? (label: 'backport requested')
* This PR touches english.txt or translations? Check the [guidelines](https://github.com/OpenTTD/OpenTTD/blob/master/docs/eints.md)
* This PR affects the save game format? (label 'savegame upgrade')
* This PR affects the GS/AI API? (label 'needs review: Script API')
    * ai_changelog.hpp, gs_changelog.hpp need updating.
    * The compatibility wrappers (compat_*.nut) need updating.
* This PR affects the NewGRF API? (label 'needs review: NewGRF')
    * newgrf_debug_data.h may need updating.
    * [PR must be added to API tracker](https://wiki.openttd.org/en/Development/NewGRF/Specification%20Status)

### CODE REVIEW

**Review by glx22 — DISMISSED** [review]

**Review by 2TallTyler — COMMENTED** [review]
Is it worth removing the string codes from the vanilla strings? (Could be a separate PR)

**Comment by PeterN** [comment]
Yes, I've also spotted another place where it needs to be fixed.

**Comment by glx22** [comment]
I'm not sure about the naming for `SetText` maybe `SetTextStyle`. For me `SetText` would mean to set the actual text (which is done by `SetDataTip`)

**Comment by stormcone** [comment]
In the `widget.cpp` please update the doxygen comments for the functions with the `fontSize` parameter.

**Comment by PeterN** [comment]
Thanks, updated comments!

**Review by glx22 — DISMISSED** [review]
Not sure how EINTS will react for translations editions (worst case it will revert or remove the strings)

File: src/widget.cpp [hunk 1]
```diff
@@ -487,14 +487,14 @@ static inline void DrawImageButtons(const Rect &r, WidgetType type, Colours colo
  * @param str     Text to draw.
  * @param align   Alignment of the text.
  */
```
Comment (glx22) [inline src/widget.cpp:490]: Missing doxygen for `fs`

File: src/widget.cpp [hunk 2]
```diff
@@ -504,11 +504,11 @@ static inline void DrawLabel(const Rect &r, WidgetType type, bool clicked, TextC
  * @param str    Text to draw.
  * @param align  Alignment of the text.
  */
```
Comment (glx22) [inline src/widget.cpp:508]: Same

File: src/widget.cpp [hunk 3]
```diff
@@ -519,10 +519,10 @@ static inline void DrawText(const Rect &r, TextColour colour, StringID str, Stri
  * @param str         Text to draw.
  * @param align       Alignment of the text.
  */
```
Comment (glx22) [inline src/widget.cpp:524]: And here too

File: src/widget.cpp [hunk 4]
```diff
@@ -673,11 +673,11 @@ static inline void DrawHorizontalScrollbar(const Rect &r, Colours colour, bool l
  * @param str         Text of the frame.
  * @param align       Alignment of the text in the frame.
  */
```
Comment (glx22) [inline src/widget.cpp:679]: And here

File: src/widget.cpp [hunk 5]
```diff
@@ -802,7 +802,7 @@ static inline void DrawCloseBox(const Rect &r, Colours colour)
  * @param str         Text to draw in the bar.
  * @param align       Alignment of the text.
  */
```
Comment (glx22) [inline src/widget.cpp:809]: Again

**Comment by PeterN** [comment]
Think I've caught all the doxygen updates now.

**Review by glx22 — APPROVED** [review]

**Pull request merged**